irohad/main/impl/on_demand_ordering_init.cpp

```diff
@@ -171,38 +172,62 @@ namespace iroha {
             proposal_factory,
         std::shared_ptr<ametsuchi::TxPresenceCache> tx_cache,
         consensus::Round initial_round) {
+      auto delay = [reject_counter = static_cast<uint64_t>(0),
+                    local_counter =
+                        static_cast<uint64_t>(0)](const auto &commit) mutable {
+        using iroha::synchronizer::SynchronizationOutcomeType;
+        if (commit.sync_outcome == SynchronizationOutcomeType::kReject
+            or commit.sync_outcome == SynchronizationOutcomeType::kNothing) {
+          if (local_counter++ == 2) {
```

Comment:
I guess it is better to move out increment operator from `if` statement.

Also, constant `2` is not so meaningful better use names constant here or describe why two is enough.